odoo/modules/neutralize.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import textwrap
+
+import odoo
+from .module import get_manifest
+
+
+def get_installed_modules(cursor):
+    cursor.execute('''
+        SELECT name FROM ir_module_module
+        WHERE state IN ('installed', 'to upgrade', 'to remove');
+    ''')
+    return [result[0] for result in cursor.fetchall()]
+
+def get_neutralization_sql(modules):
+    queries = []
+
+    # neutralization for each module
+    for module in modules:
+        manifest = get_manifest(module)
+        if manifest and manifest['neutralize']:
+            filename = odoo.modules.get_module_resource(module, manifest['neutralize'])
+            with odoo.tools.misc.file_open(filename) as file:
+                queries.append(file.read())
+
+    # neutralization flags for the database
+    queries.append(textwrap.dedent('''\
+        INSERT INTO ir_config_parameter(key, value)
+        VALUES ('database.is_neutralized', true)
+        ON CONFLICT (key)
+        DO UPDATE SET value = true;
+    '''))
+
+    # neutralization watermarks
+    queries.append(textwrap.dedent('''\
+        UPDATE ir_ui_view
+        SET active=true
+        WHERE key IN ('web.neutralize_banner', 'website.neutralize_ribbon');
+    '''))
+
+    return '\n\n'.join(queries)
```

Comment:
As pointed out by Raphael, it's better to ensure all the queries end with a semicolon.
```suggestion
    return ';\n\n'.join(q.rstrip(';') for q in queries) + ';'
```